### PR TITLE
Fix SQL syntax error that happens when scanning a table without clustering key in JDBC adapter

### PR DESCRIPTION
**Pull request by brfrn169**
This PR is a fix for a bug where an SQL syntax error happens when scanning a table without clustering key in JDBC adapter. We basically don't need to scan a table without clustering key, but an error should not happen if we do that. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java [hunk 1]
```diff
@@ -86,7 +86,9 @@ private String conditionSqlString() {
   }
 
   private String orderBySqlString() {
-    if (!isRangeQuery || indexedColumn.isPresent()) {
+    if (!isRangeQuery
+        || indexedColumn.isPresent()
+        || tableMetadata.getClusteringKeyNames().isEmpty()) {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java:91]: We always added ORDER BY clause when we scan a table (except index scan) but if the table doesn't have clustering key, we should not have done that.

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
(Should the PR title be changed to `Fix ...` ?)

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**